### DISCUSSION TITLE
Run and Debug buttons and keyboard shortcuts are ignoring system path

### DISCUSSION THREAD

**Discussion by pbaret** [microsoft/vscode-cmake-tools#3204] [Q&A]
The extensions buttons are ignoring my system Path so I am not able to run without debugging or debug a c++ program which is using external library from my system PATH using the buttons.

![image](https://github.com/microsoft/vscode-cmake-tools/assets/16573776/4e9781bb-459d-4508-85b3-21fd4928a900)

The F5 and CTRL+F5 shortcuts don't seem to work either WHEREAS (and this is very weird) clicking from the menu works 
![image](https://github.com/microsoft/vscode-cmake-tools/assets/16573776/6a9fb282-f56c-4455-ab22-b194d7d9c9a1)

Any idea how to configure that?

**Comment by gcampbell-msft** [comment]
@pbaret What do you mean "they are ignoring your system PATH"? Did you open VS Code with this system path?

**Comment by florenso** [comment]
its a bit strange you can launch with cmake. but cmakes quick launch buttons do not include the PATH or any other args.

use launch.json instead.

see #3989

**Comment by gcampbell-msft** [comment]
@florenso This sounds like a bug, or at least something we should think about improving, could you create an issue for this?

**Reply by florenso** [reply]
I did 
#3990

**Comment by pbaret** [comment]
@gcampbell-msft I mean it fails launching the program because it acts like it can't find the external library (dll files for instance, which I am 100% sure are in the "system path" variable of my computer).
